packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.spec.js

```diff
@@ -0,0 +1,232 @@
+        {
+          key: 'column1',
+          label: 'Column 1',
+        },
+      ],
+    });
+
+    render(<ColumnManager {...props} />);
+
+    expect(screen.getByText(/Column 1/i)).toBeInTheDocument();
+  });
+
+  it('should render visible columns', () => {
+    const props = createTestProps({
+      selectedColumns: [
+        {
+          key: 'column1',
+          label: 'Column 1',
+        },
+      ],
+      availableColumns: [
+        {
+          key: 'column1',
+          label: 'Column 1',
+        },
+      ],
+    });
+
+    render(<ColumnManager {...props} />);
+
+    expect(screen.getByText(/Column 1/i)).toBeInTheDocument();
+  });
+
+  describe('when `disableHiddenColumnSearch` is true', () => {
+    it('should render columns search input', () => {
+      const props = createTestProps({
+        selectedColumns: [
+          {
+            key: 'column1',
+            label: 'Column 1',
+          },
+        ],
+        availableColumns: [
+          {
+            key: 'column1',
+            label: 'Column 1',
+          },
+        ],
+        disableHiddenColumnSearch: false,
+        searchHiddenColumnsPlaceholder: 'search',
+        searchHiddenColumns: jest.fn(),
+      });
+
+      render(<ColumnManager {...props} />);
+
+      expect(screen.getByText(/search/i)).toBeInTheDocument();
+    });
+  });
+
+  describe('buttons', () => {
+    it('primary button', () => {
+      const primaryButtonLabel = 'Primary Test';
+      const props = createTestProps({
+        primaryButton: (
+          <PrimaryButton label={primaryButtonLabel} onClick={jest.fn()} />
+        ),
+      });
+      render(<ColumnManager {...props} />);
+      expect(screen.getByText(primaryButtonLabel)).toBeInTheDocument();
+    });
+
+    it('secondary button', () => {
+      const secondaryButtonLabel = 'Secondary Test';
+      const props = createTestProps({
+        secondaryButton: (
+          <SecondaryButton label={secondaryButtonLabel} onClick={jest.fn()} />
+        ),
+      });
+      render(<ColumnManager {...props} />);
+      expect(screen.getByText(secondaryButtonLabel)).toBeInTheDocument();
+    });
+  });
+
+  describe('handleDragEnd', () => {
+    describe('when destination and source is the hidden columns panel', () => {
+      it('should not call `onUpdateColumns`', () => {
+        const onUpdateColumns = jest.fn();
+        const setIsDragging = jest.fn();
+        const props = createTestProps({
+          selectedColumns: [
+            {
+              key: 'column1',
+              label: 'Column 1',
+            },
+          ],
+          availableColumns: [
+            {
+              key: 'column1',
+              label: 'Column 1',
+            },
+          ],
+          onUpdateColumns,
+        });
+
+        const dragResult = {
+          destination: { droppableId: HIDDEN_COLUMNS_PANEL },
+          source: { droppableId: HIDDEN_COLUMNS_PANEL },
+        };
+
+        handleDragEnd(
```

Comment:
Is it possible to use RTL to test the drag/drop behavior, instead of testing the function itself?

Reply:
I've tried that using the `fireEvent` but couldn't come up with a way to make it work